### PR TITLE
ability to save books added

### PR DESCRIPTION
**Pull request by Conartisttt**

### CODE REVIEW

File: client/src/pages/Search.jsx
Comment (Conartisttt) [inline client/src/pages/Search.jsx:1]: Add save book mutation to button.
Save book to user on click

File: client/src/utils/mutations.js
Comment (Conartisttt) [inline client/src/utils/mutations.js:1]: Update mutation to save book with updated models.

File: server/.gitignore
Comment (Conartisttt) [inline server/.gitignore:1]: Added a new gitignore to try to get my node_modules to not track. I think it worked?

File: client/src/utils/queries.js
Comment (Conartisttt) [inline client/src/utils/queries.js:1]: update query to match model

File: server/models/Book.js
Comment (Conartisttt) [inline server/models/Book.js:1]: Attempt to have bookId be unique so that it does not double save

File: server/schemas/resolvers.js
Comment (Conartisttt) [inline server/schemas/resolvers.js:1]: update resolvers for book on user model, not savedBooks

File: server/schemas/typeDefs.js
Comment (Conartisttt) [inline server/schemas/typeDefs.js:1]: update typedefs to reflect changes on model structure

**Review by dhiduchick — APPROVED** [review]
lgtm!

**Pull request merged**